sdk/core/core-https/src/pipelineRequest.ts

```diff
@@ -130,7 +130,7 @@ class PipelineRequestImpl implements PipelineRequest {
     this.keepAlive = options.keepAlive ?? true;
     this.proxySettings = options.proxySettings;
     this.skipDecompressResponse = options.skipDecompressResponse ?? false;
-    this.streamResponseBody = options.streamResponseBody ?? false;
+    this.streamResponseStatusCodes = options.streamResponseStatusCodes ?? new Set<number>();
```

Comment:
why initialize it to an empty set? Shouldn't we leave it undefined since it's optional?

Reply:
Somehow I defined it as non-optional. Making it optional now.